### DISCUSSION TITLE
Is Advanced Foundry Course mandatory for Smart Contract Security Course ?

### DISCUSSION THREAD

**Discussion by Sauron-sol** [Cyfrin/security-and-auditing-full-course-s23#225] [Q&A] [answered]
Hi there,
Just completed the "Foundry Fundamentals" course and was wondering if anyone knows if the "Advanced Foundry" course is a requirement for the Smart Contract Security course? I've got skills in C, C++, Python, CI/CD... and Blockchain basics from 2020 (DeFi, Spot, Futures...).
Thanks!

**Comment by EngrPips** [comment, marked as answer]
Hello @SpectraD, Ideally, I will advise that you do the `Advance Foundry` section before going to the `Security` course, but if you are pressed for time, you can jump straight to the `Security` course. Considering all the knowledge you claim to have, you shouldn't struggle too much, and if need be, you can always touch back on the `Advance Foundry` course to clarify things.
